hpx/runtime/actions/basic_action.hpp

```diff
@@ -180,108 +263,43 @@ namespace hpx { namespace actions
-            }
-
-            template <typename ...Ts>
-            HPX_FORCEINLINE result_type invoke(std::false_type,
-                naming::address::address_type lva,
-                naming::address::component_type comptype, Ts&&... vs) const
-            {
-                return Derived::invoke(lva, comptype, std::forward<Ts>(vs)...);
-            }
-        };
+            using is_void = typename std::is_void<R>::type;
+            return invoker_impl(is_void{}, lva, comptype,
+                std::forward<Ts>(vs)...);
+        }
 
     protected:
-        /// The \a thread_function will be registered as the thread
-        /// function of a thread. It encapsulates the execution of the
-        /// original function (given by \a func).
-        struct thread_function
+        template <typename ...Ts>
+        HPX_FORCEINLINE static remote_result_type
+        invoker_impl(std::true_type,
+            naming::address::address_type lva,
+            naming::address::component_type comptype, Ts&&... vs)
         {
-            thread_function()
-            {}
-
-            thread_function(naming::id_type const& target)
-              : target_(target)
-            {}
-
-            template <typename ...Ts>
-            HPX_FORCEINLINE threads::thread_result_type
-            operator()(naming::address::address_type lva,
-                naming::address::component_type comptype, Ts&&... vs) const
-            {
-                try {
-                    LTM_(debug) << "Executing "
-                        << Derived::get_action_name(lva) << ".";
-
-                    // call the function, ignoring the return value
-                    Derived::invoke(lva, comptype, std::forward<Ts>(vs)...);
-                }
-                catch (hpx::thread_interrupted const&) { //-V565
-                    /* swallow this exception */
-                }
-                catch (std::exception const& e) {
-                    LTM_(error)
-                        << "Unhandled exception while executing "
-                        << Derived::get_action_name(lva) << ": " << e.what();
-
-                    // report this error to the console in any case
-                    hpx::report_error(std::current_exception());
-                }
-                catch (...) {
-                    LTM_(error)
-                        << "Unhandled exception while executing "
-                        << Derived::get_action_name(lva);
-
-                    // report this error to the console in any case
-                    hpx::report_error(std::current_exception());
-                }
-
-                // Verify that there are no more registered locks for this
-                // OS-thread. This will throw if there are still any locks
-                // held.
-                util::force_error_on_lock();
-                return threads::thread_result_type(threads::terminated,
-                    threads::invalid_thread_id);
```

Comment:
What happened to the exception handling?

Reply:
The nested `thread_function` class moved to namespace scope, and now resides at lines 88:149

Reply:
Thanks, missed that!